include/agent.h

```diff
@@ -0,0 +1,73 @@
+#ifndef AGENT_CPP_H
+#define AGENT_CPP_H
+
+#include <vector>
+
+#include "directive.h"
+
+
+namespace Rep
+{
+
+    class Agent
+    {
+    public:
+        /* The type for the delay. */
+        typedef float delay_t;
```

Comment:
I tend to do this if I'm going to use that type in more than one place. If everyone thinks it's clunky, I don't feel strongly about this instance.